.github/workflows/ci_pr.yml

```diff
@@ -0,0 +1,140 @@
+name: CI Unit tests
+
+on:
+  push:
+    branches: [ "*" ]
+  pull_request:
+    branches: [ "*" ]
+
+jobs:
+  legacy:
+    
+    name: "Python 2.6 Unit Tests"
+    runs-on: ubuntu-16.04
+
+    strategy:
+      fail-fast: false
+    env:
+      NOSEOPTS: "--verbose"
+
+    steps:
+
+    - name: Install Python 2.6
+      run: |
+        sudo add-apt-repository ppa:deadsnakes/ppa -y
```

Comment:
can we do 'set -euo pipefail' on those scripts (is this bash?)

Reply:
I think `runs-on: ubuntu-16.04` is actually using bash. It's hard to find concrete info on it online, but `command not found` errors for 18.04 (during testing of this PR) show that `bash` is being used for `ubuntu-18.04` run commands... And I think plain `sh` is being used for 14.04. I can't remember, however, what 16.04 is using, but I think it was bash as well.

I can obviously just test it via another push. I am hesitant to use anything other than purely POSIX-compliant commands, however, as I can't find any documentation guaranteeing that, e.g., `bash` is used.

Reply:
Follow up: ubuntu-16.04 does in fact use bash for any `run` actions.

Reply:
We should mirror what we do for Travis at least:
`set -u`

Reply:
But on that note, would this `run` step fail if any command in this shell fail without `set -euo pipefail`? If not, then we should add it

Reply:
any failed command in a step do in fact fail that entire step!